src/screens/ReferFriends/ReferralContacts.js

```diff
@@ -0,0 +1,333 @@
+import Toast from 'components/Toast';
+
+import { fontStyles, spacing } from 'utils/variables';
+import { getRemainingDailyInvitations } from 'utils/referrals';
+import { isValidPhone } from 'utils/validators';
+
+import type { NavigationScreenProp } from 'react-navigation';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import type { ReferralContact, SentInvitationsCount } from 'reducers/referralsReducer';
+
+import { setContactsForReferralAction } from 'actions/referralsActions';
+import { fetchPhoneContactsAction } from 'actions/phoneContactsActions';
+
+import { ADD_EDIT_USER, REFER_MAIN_SCREEN } from 'constants/navigationConstants';
+import { ALLOWED_DAILY_INVITES } from 'constants/referralsConstants';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  setContactsForReferral: (contacts: ReferralContact[]) => void,
+  addedContactsToInvite: ReferralContact[],
+  phoneContacts: ReferralContact[],
+  isFetchingPhoneContacts: boolean,
+  isFetchingPhoneContactsComplete: boolean,
+  phoneContactsFetchError: boolean,
+  fetchPhoneContacts: () => void,
+  isEmailVerified: boolean,
+  isPhoneVerified: boolean,
+  alreadyInvitedContacts: ReferralContact[],
+  sentInvitationsCount: SentInvitationsCount,
+  userPhone: string,
+  userEmail: string,
+};
+
+type State = {
+  query: string,
+  selectedContacts: ReferralContact[],
+};
+
+
+const EmptyStateWrapper = styled.View`
+  width: 100%;
+  align-items: center;
+  margin: 20px 0 30px;
+`;
+
+const ButtonWrapper = styled.View`
+  justify-content: center;
+  padding: ${spacing.layoutSides}px;
+`;
+
+const StyledTextLink = styled(TextLink)`
+  ${fontStyles.regular};
+`;
+
+const MIN_QUERY_LENGTH = 3;
+
+const getInfoTypes = (isPhoneVerified, isEmailVerified) => {
+  // one or another should be true, otherwise feature is not available at all
+  if (!isPhoneVerified) {
+    return 'phone';
+  } else if (!isEmailVerified) {
+    return 'email';
+  }
+  return null;
+};
+
+const getFilteredContacts = (contacts, _query: string) => {
+  if (!_query || _query.length < MIN_QUERY_LENGTH) return contacts;
+  const query = _query.toUpperCase();
+  return contacts.filter(({ name, email = '', phone = '' }) => {
+    return name.toUpperCase().indexOf(query) > -1
+      || email.toUpperCase().indexOf(query) > -1
+      || phone.indexOf(query) > -1;
+  });
+};
+
+class ReferralContacts extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.handleSearch = debounce(this.handleSearch, 500);
+    this.state = {
+      query: '',
+      selectedContacts: props.addedContactsToInvite || [],
+    };
+  }
+
+  componentDidMount() {
+    const {
+      isFetchingPhoneContactsComplete,
+      isFetchingPhoneContacts,
+      fetchPhoneContacts,
+    } = this.props;
+
+    if (!isFetchingPhoneContacts && !isFetchingPhoneContactsComplete) {
+      fetchPhoneContacts();
+    }
+  }
+
+
+  handleSearch = (query: any) => {
```

Comment:
why `any`?

Reply:
Updated to `string`, also updated the component to match, as it seems like it's always gonna be a string.